tests/lib/bin/retry-tool

```diff
@@ -40,33 +40,42 @@ attempt. On failure the exit code from the final attempt is returned.
         default=1,
         help="grace period between attempts (default %(default)ss)",
     )
+    parser.add_argument(
+        "--quiet",
+        dest="verbose",
+        action="store_false",
+        default=True,
+        help="refrain from printing any output",
+    )
     parser.add_argument(
         "cmd", metavar="COMMAND", nargs="...", help="command to execute"
     )
     return parser
 
 
-def run_cmd(cmd, n, wait):
-    # type: (List[Text], int, float) -> int
+def run_cmd(cmd, n, wait, verbose):
+    # type: (List[Text], int, float, bool) -> int
     retcode = 0
     for i in range(1, n + 1):
         retcode = subprocess.call(cmd)
         if retcode == 0:
             break
-        print(
-            "retry: command {} failed with code {}".format(" ".join(cmd), retcode),
-            file=sys.stderr,
-        )
-        if i < n:
+        if verbose:
             print(
-                "retry: next attempt in {} second(s) (attempt {} of {})".format(
-                    wait, i, n
-                ),
+                "retry: command {} failed with code {}".format(" ".join(cmd), retcode),
                 file=sys.stderr,
             )
+        if i < n:
+            if verbose:
+                print(
+                    "retry: next attempt in {} second(s) (attempt {} of {})".format(
+                        wait, i, n
+                    ),
+                    file=sys.stderr,
+                )
             time.sleep(wait)
     else:
-        if n > 1:
+        if verbose and n > 1:
```

Comment:
We expected the command to succeed in n attempts, so reaching here is unexpected. Wonder if we should print the message anyway. Makes the switch named `--quiet` a bit akward, so maybe `--no-log` then?

Reply:
I was thinking about it as well, oh oh :) Such a hard choice.

So here's my proposal instead. I was thinking about this earlier but decided not to do it because YAGNI but this shows it may have been useful:

The `-n` switch can now be both a number and a range `n-m`. When it is a range it indicates that we expect the first `n` attempts to fail and those automatically become quiet. After those the remaining up-to-`m` attempts log as usual.

Reply:
Sounds like a followup material.

Reply:
Yeah, I'm happy to iterate on this some more. We can always respect `--quiet` even if we have the range syntax implemented.